### ISSUE TITLE
PollingDirectoryMonitor.poll NullPointerException

### ISSUE THREAD

**Issue by cosmin33** [jcustenborder/kafka-connect-spooldir#15]
**Stacktrace:**
[2017-01-18 18:25:48,139] INFO Opening /home/cosmo/devel/syncrasy/KafkaConnectExample/input/temp.csv.PROCESSING (io.confluent.kafka.connect.source.io.PollingDirectoryMonitor:216)
[2017-01-18 18:25:48,168] ERROR Exception encountered processing line 1 of /home/cosmo/devel/syncrasy/KafkaConnectExample/input/temp.csv.PROCESSING. (io.confluent.kafka.connect.source.io.PollingDirectoryMonitor:231)
org.apache.kafka.connect.errors.ConnectException: java.lang.NullPointerException: SchemaConfig.name cannot be null
	at io.confluent.kafka.connect.source.io.PollingDirectoryMonitor.poll(PollingDirectoryMonitor.java:221)
	at io.confluent.kafka.connect.source.SpoolDirectoryTask.poll(SpoolDirectoryTask.java:51)
	at org.apache.kafka.connect.runtime.WorkerSourceTask.execute(WorkerSourceTask.java:155)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:140)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:175)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException: SchemaConfig.name cannot be null
	at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:226)
	at io.confluent.kafka.connect.source.io.processing.csv.SchemaConfig.parserConfigs(SchemaConfig.java:42)
	at io.confluent.kafka.connect.source.io.processing.csv.CSVRecordProcessor.configure(CSVRecordProcessor.java:150)
	at io.confluent.kafka.connect.source.io.PollingDirectoryMonitor.poll(PollingDirectoryMonitor.java:219)
	... 9 more
[2017-01-18 18:25:48,170] INFO Closing /home/cosmo/devel/syncrasy/KafkaConnectExample/input/temp.csv.PROCESSING (io.confluent.kafka.connect.source.io.PollingDirectoryMonitor:130)
[2017-01-18 18:25:48,170] ERROR Error during processing, moving /home/cosmo/devel/syncrasy/KafkaConnectExample/input/temp.csv.PROCESSING to /home/cosmo/devel/syncrasy/KafkaConnectExample/error. (io.confluent.kafka.connect.source.io.PollingDirectoryMonitor:139)

**Configuration (with REST)**
{
	"name": "CsvSourceTest",
	"config": {
		"connector.class": "io.confluent.kafka.connect.source.SpoolDirectoryConnector",
		"tasks.max": "1",
		"topics": "connect-test",
		"record.processor.class": "io.confluent.kafka.connect.source.io.processing.csv.CSVRecordProcessor",
		"input.file.pattern": "^.*\\.csv$",
		"finished.path": "/home/cosmo/devel/syncrasy/KafkaConnectExample/finished",
		"error.path": "/home/cosmo/devel/syncrasy/KafkaConnectExample/error",
		"input.path": "/home/cosmo/devel/syncrasy/KafkaConnectExample/input",
		"halt.on.error": false,
		"topic": "connect-test",
		"include.file.metadata": false,
		"batch.size": 1000,
		"csv.null.field.indicator": "BOTH",
		"csv.parser.timestamp.date.formats": "yyyy-MM-dd'T'HH:mm:ss'Z'",
		"csv.first.row.as.header": true,
		"csv.schema": "{\"keys\":[\"id\"],\"fields\":[{\"name\":\"id\",\"type\":\"int32\",\"required\":true},{\"name\":\"name\",\"type\":\"string\",\"required\":true}]}"
	}
}

**When I try to copy this file in the input directory:**
**File "temp.txt"**
id,name
1,cosmo
2,steve
3,alin
4,gavin
5,leo

**Comment by jcustenborder** [comment]
It's complaining because the schema name is null. It's a little buried in the error message.

Try changing your `csv.schema` to include a `name` property in the json. Something like this:

```json
{
  "csv.schema": "{"name":"com.example.foo.User","keyName":"com.example.foo.UserKey","keys":["id"],"fields":[{"name":"id","type":"int32","required":true}
}
``` 

`name` is the property for the value schema.
`keyName` is the property for the key schema.

This needs to be documented better.

**Comment by jcustenborder** [comment]
This was fixed with #17.
